### PR TITLE
Use match for prover query.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/chriseth/powdr/issues/163

### CODE REVIEW

**Review by leonardoalt — COMMENTED** [review]
Why doesnt this have changes in lang and tests?

**Comment by chriseth** [comment]
It changes the way the prover query is performed. The tests are purely semantic and still work as is. There is one test expectation change, though. What do you mean by "lang"?

**Comment by leonardoalt** [comment]
Ah ok, I somehow thought this would change the way you query things in the syntax too

File: src/witness_generator/generator.rs [hunk 1]
```diff
@@ -279,10 +279,29 @@ where
...
+            .evaluate(scrutinee, EvaluationRow::Next)?
+            .constant_value()
+            .ok_or_else(|| "Match scrutinee not constant".to_string())?;
+        let (_, expr) = arms
+            .iter()
+            .find(|(n, _)| n.is_none() || n.as_ref() == Some(&v))
+            .ok_or_else(|| format!("Match arm not found for value {v}"))?;
+        self.interpolate_query(expr).map_err(|e| e.into())
+    }
+
```
Comment (leonardoalt) [inline src/witness_generator/generator.rs:304]: Wasnt this part of another PR?

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**